File: 4.8-bc/py/stage-3-hexagonal/app/commerce/infra/postgres/order_repo.py
"""Postgres adapter for Commerce Repository."""
from __future__ import annotations

from sqlalchemy import text
from sqlalchemy.ext.asyncio import async_sessionmaker

from app.commerce.domain.order import Order


class OrderRepo:
    def __init__(self, session_maker: async_sessionmaker) -> None:
        self._session_maker = session_maker

    async def create(self, o: Order) -> None:
        async with self._session_maker() as session:
            await session.execute(
                text(
                    "INSERT INTO commerce_orders "
                    "(id, user_id, total_cents, status, created_at) "
                    "VALUES (:id, :user_id, :total_cents, :status, :created_at)"
                ),
                {
                    "id": o.id,
                    "user_id": o.user_id,
                    "total_cents": o.total_cents,
                    "status": o.status,
                    "created_at": o.created_at,
                },
            )
            await session.commit()

File: 4.8-bc/ts/stage-3-hexagonal/src/commerce/infra/postgres/orderRepo.ts
import type { DbPool } from "../../../shared/db.js";
import type { Order } from "../../domain/order.js";
import type { Repository } from "../../domain/repository.js";

export class PgOrderRepo implements Repository {
  constructor(private readonly db: DbPool) {}

  async create(o: Order): Promise<void> {
    await this.db.query(
      `INSERT INTO commerce_orders (id, user_id, total_cents, status, created_at)
       VALUES ($1, $2, $3, $4, $5)`,
      [o.id, o.userId, o.totalCents, o.status, o.createdAt],
    );
  }
}
